pynbody/snapshot/gadgethdf.py

```diff
@@ -563,9 +563,18 @@ def __init_unit_information(self):
 
         cosmo = 'HubbleParam' in list(self._get_hdf_parameter_attrs().keys())
         if cosmo:
-            for fac in self._get_cosmo_factors(self._hdf_files[0], 'Coordinates'): dist_unit *= fac
-            for fac in self._get_cosmo_factors(self._hdf_files[0], 'Velocities'): vel_unit *= fac
-            for fac in self._get_cosmo_factors(self._hdf_files[0], 'Mass'): mass_unit *= fac
+            try:
+                for fac in self._get_cosmo_factors(self._hdf_files[0], 'Coordinates'): dist_unit *= fac
+            except:
```

Comment:
Please change this bare except to a more specific exception e.g. `except KeyError`, if that's the appropriate one

Reply:
I agree to update all the excepts.